### ISSUE TITLE
_step in generally is called more times after eof_have_data

### ISSUE THREAD

**Issue by danielniccoli** [cherokee/webserver#477]
_Original author: ste...@konink.de (May 23, 2009 13:00:55)_

<b>What steps will reproduce the problem?</b>
1. take a client with a small buffer, or a big document
2. review that the first time _step is called, and it return eof_have_data
(aka everything is produced just send)
3. at the sending phase where we get due to this return value, our return
value is overridden by the return value of the send function. This will be
(in worst case: ret_eagain).
4. Eagain triggers the logic to take another go at the _step function,
basically creating unexpected results.
5. This goes wrong in dbslayer for large requests

A suggestion patch is attached as reference. And implements a 'have all
data' and do not _step anymore logic.

_Original issue: http://code.google.com/p/cherokee/issues/detail?id=474_
